frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx

```diff
@@ -0,0 +1,190 @@
+  const [insecureTraffic, setInsecureTraffic] = React.useState(prefferedInsecureTraffic);
+
+  const tlsTerminationSelectOptions: JSX.Element[] = React.useMemo(
+    () =>
+      Object.keys(TerminationTypes).map((tlsTerminationOption) => (
+        <SelectOption key={tlsTerminationOption} value={tlsTerminationOption}>
+          {TerminationTypes[tlsTerminationOption]}
+        </SelectOption>
+      )),
+    [],
+  );
+
+  const insecureTrafficSelectOptions: JSX.Element[] = React.useMemo(
+    () =>
+      Object.keys(InsecureTrafficTypes).map((insecureTrafficOption) => (
+        <SelectOption key={insecureTrafficOption} value={insecureTrafficOption}>
+          {InsecureTrafficTypes[insecureTrafficOption]}
+        </SelectOption>
+      )),
+    [],
+  );
+
+  const passthroughInsecureTrafficTypesSelectOptions: JSX.Element[] = React.useMemo(
+    () =>
+      Object.keys(PassthroughInsecureTrafficTypes).map((passthroughInsecureTraffic) => (
+        <SelectOption key={passthroughInsecureTraffic} value={passthroughInsecureTraffic}>
+          {PassthroughInsecureTrafficTypes[passthroughInsecureTraffic]}
+        </SelectOption>
+      )),
+    [],
+  );
+
+  const onSecureRouteChecked = React.useCallback(
+    (checked: boolean) => {
+      setSecureRouteChecked(checked);
+      setPreferredRoutingOptions({
+        secure: checked,
+        tlsTermination,
+        insecureTraffic,
+      });
+    },
+    [insecureTraffic, setPreferredRoutingOptions, tlsTermination],
+  );
+
+  const onTLSTerminationToggle = React.useCallback(
+    (isOpen: boolean) => setIsTLSTerminationOpen(isOpen),
+    [],
+  );
+
+  const onInsecureTrafficToggle = React.useCallback(
+    (isOpen: boolean) => setIsInsecureTrafficOpen(isOpen),
+    [],
+  );
+
+  const onTLSTerminationSelect = (_, selection: string) => {
+    setTlsTermination(selection);
+    setPreferredRoutingOptions({
+      secure: secureRouteChecked,
+      tlsTermination: selection,
+      insecureTraffic,
+    });
+    setIsTLSTerminationOpen(false);
+  };
+
+  const onInsecureTrafficSelect = (_, selection: string) => {
+    setInsecureTraffic(selection);
+    setPreferredRoutingOptions({
+      secure: secureRouteChecked,
+      tlsTermination,
+      insecureTraffic: selection,
+    });
+    setIsInsecureTrafficOpen(false);
+  };
+
+  return preferredRoutingOptionsLoaded ? (
+    <>
+      <Checkbox
+        id="secure-route-checkbox"
+        label={t('devconsole~Secure route')}
+        isChecked={secureRouteChecked}
+        data-checked-state={secureRouteChecked}
+        onChange={onSecureRouteChecked}
+        aria-label={t('devconsole~Secure route')}
+        description={t(
+          'devconsole~Routes can be secured using several TLS termination types for serving certificates.',
+        )}
+        className="odc-secure-route-fields__secure-route"
+      />
+      {secureRouteChecked && (
+        <>
```

Comment:
As discussed in our PM/dev sync we agreed that we want always show the Form Groups, so that the admin can configure the TLS termination and Insecure Traffic options also when the Secret default is disabled. (The user can enable it and can use the custom options in this case.)